src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -1330,23 +1318,33 @@ private void AbortStreams(Exception abortException)
             // we could hold pool lock while trying to grab connection lock in Dispose().
             _pool.InvalidateHttp2Connection(this);
 
+            List<Http2Stream> streamsToAbort = new List<Http2Stream>();
+
             lock (SyncObject)
             {
                 if (NetEventSource.IsEnabled) Trace($"{nameof(abortException)}={abortException}");
```

Comment:
Should this be moved to the beginning of the method, prior to the InvalidateHttp2Connection call?  I don't remember why we felt it was important to call it under the lock, but it seems misleading here for someone reading the trace.

Reply:
Will do